### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mighty-joey**
Made some changes to make contributing to this project a bit more approachable for the newbie open source developer.

### CODE REVIEW

**Comment by marek-lach** [comment]
Such a guide should probably also include steps on how to create a fork in the first place, and also how to create a branch in your own repository to contribute to?

**Review by mighty-joey — COMMENTED** [review]
Updated to add additional info

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,6 +12,20 @@ You can contribute in the following ways:
...
+We want all members of the GitHub community to feel welcome when it comes to contributing to our project. You can follow these step-by-step instructions to make a contribution via a pull request.
+
+1. Fork a repository from Mastodon and create a branch to build new features or test out ideas - instructions available here: https://help.github.com/en/articles/fork-a-repo
+2. Navigate to the original repository you created your fork from.
+3. To the right of the Branch menu, click New pull request.
+4. On the Compare page, click compare across forks.
+5. Confirm that the base fork is the repository you'd like to merge changes into. Use the base branch drop-down menu to select the branch of the upstream repository you'd like to merge changes into.
+6. Use the head fork drop-down menu to select your fork, then use the compare branch drop-down menu to select the branch you made your changes in.
+7. Type a title and description for your pull request.
+8. If you do not want to allow anyone with push access to the upstream repository to make changes to your PR, unselect Allow edits from maintainers.
```
Comment (nightpool) [inline CONTRIBUTING.md]: the distinction here is confusing and probably not particularly useful in a entry-level tutorial like this one

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,6 +12,20 @@ You can contribute in the following ways:
 
 If your contributions are accepted into Mastodon, you can request to be paid through [our OpenCollective](https://opencollective.com/mastodon).
 
+## New to Open Source? START HERE
+
+We want all members of the GitHub community to feel welcome when it comes to contributing to our project. You can follow these step-by-step instructions to make a contribution via a pull request.
+
+1. Fork a repository from Mastodon and create a branch to build new features or test out ideas - instructions available here: https://help.github.com/en/articles/fork-a-repo
```
Comment (nightpool) [inline CONTRIBUTING.md]: These instructions are really long and there's a much easier way to make pull requests by using the Github web editor for small changes.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,6 +12,20 @@ You can contribute in the following ways:
...
+2. Navigate to the original repository you created your fork from.
+3. To the right of the Branch menu, click New pull request.
+4. On the Compare page, click compare across forks.
+5. Confirm that the base fork is the repository you'd like to merge changes into. Use the base branch drop-down menu to select the branch of the upstream repository you'd like to merge changes into.
+6. Use the head fork drop-down menu to select your fork, then use the compare branch drop-down menu to select the branch you made your changes in.
```
Comment (nightpool) [inline CONTRIBUTING.md]: assuming someone knows what a branch is or how to make changes on it doesn't seem to make sense for this level of introductory guide?

**Review by nightpool — CHANGES_REQUESTED** [review]
Hey! Thanks for the contribution. I agree that we need more detailed and approachable instructions on how to contribute changes!

Left a couple of comments about specifics but overall i'm kind of confused about who the target audience is for this step-by-step guide. Someone who's familiar with a feature-branch-based git workflow but has never used github before? Someone who can install, make changes to, and test the rails app but hasn't used git? It assumes that the reader will already be familiar with extremely crucial steps like "how to run the app" and "how to push changes to a branch"—or even "what a branch is". I'm not sure there's an audience that understands all of these concepts but needs a step-by-step guide on how to make a pull request. 

Also, it looks like these instructions were just taken wholesale from https://help.github.com/en/articles/creating-a-pull-request-from-a-fork? That's a little bit sketchy.

I would suggest re-writing this section to cover proposing changes using the Github web editor, which is much easier for new users, or maybe just linking to https://help.github.com/en/articles/editing-files-in-another-users-repository

Thank you for taking the time and initiative to start this conversation! I would love to see an improvement in mastodon's getting started documentation for contributing

**Comment by mighty-joey** [comment]
Thanks for considering. I proposed these changes for a newcomer like myself. I am new to GitHub, and a CS student. This was a requirement for a course I am taking.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Gargron) [inline CONTRIBUTING.md]: Maybe this link could be like `([See instructions](https://help.github.com/en/articles/fork-a-repo))` for brevity?

File: CONTRIBUTING.md (hunk 1 above)
Comment (Gargron) [inline CONTRIBUTING.md]: I feel like this line is not necessary.

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by nightpool** [comment]
closed due to inactivity

**Pull request closed without merging**